### PR TITLE
Ticket sale: implement ticket check

### PR DESCRIPTION
**Pull request by velev8**
Added a QR Code scan and ability to acept or deny what is sent back by the server. 

Still planning on improving the UI and show more details (like validity of the ticket) by using colors

#806

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/TicketScanValidationActivity.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.component.other.generic.activity.BaseActivity;
+
+public class TicketScanValidationActivity extends BaseActivity {
+
```
Comment (schulzalexander) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/TicketScanValidationActivity.java]: Is this class used at all? If not, please delete the class and the entry in the manifest.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/ConfirmCheckInFragment.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import android.widget.TextView;
+
+import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.api.app.TUMCabeClient;
+import de.tum.in.tumcampusapp.component.ui.ticket.model.TicketValidityResponse;
+import de.tum.in.tumcampusapp.utils.Utils;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
```
Comment (schulzalexander) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/ConfirmCheckInFragment.java]: Please delete unused imports.

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -738,6 +738,18 @@
...
+            android:name=".component.ui.ticket.TicketScanActivity"
+            android:configChanges="orientation|screenSize"
+            android:label="@string/app_name"
+            android:launchMode="singleTop"
+            android:parentActivityName=".component.ui.ticket.TicketScanActivity"
+            android:theme="@style/AppThemeFullscreen">
+            <meta-data
+                android:name="android.support.PARENT_ACTIVITY"
+                android:value="de.tum.in.tumcampusapp.component.ui.ticket.TicketScanActivity" />
+        </activity>
```
Comment (schulzalexander) [inline app/src/main/AndroidManifest.xml]: When the device is rotated, a grey sidebar appears:
bildschirmfoto 2018-06-12 um 17 10 35
Does this also appear on a real device? If so, I think we could lock the screen to portrait mode for the validity scan.

**Comment by velev8** [comment]
Thanks, I will work on locking the screen so the grey bar doesn't show up and include a tool bar on top!

**Comment by kordianbruck** [comment]
@velev8 do you want to merge this and work on separate issue branches then?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -638,4 +642,5 @@ Signature: %5$s</string>
     <string name="discard">Discard</string>
     <string name="keep_editing">Keep editing</string>
 
+>>>>>>> feature/ticketsale
```
Comment (pfent) [inline app/src/main/res/values/strings.xml]: This is probably a merge gone wrong?

File: app/build.gradle [hunk 5]
```diff
@@ -102,8 +102,9 @@ dependencies {
     // Android arch components
     implementation "android.arch.persistence.room:rxjava2:$roomVersion"
     implementation "android.arch.persistence.room:runtime:$roomVersion"
-    implementation 'android.arch.lifecycle:runtime:1.1.1'
-    implementation 'android.arch.lifecycle:extensions:1.1.1'
+    implementation "android.arch.lifecycle:runtime:1.1.1"
+    implementation "android.arch.lifecycle:extensions:1.1.1"
+    implementation 'com.android.support:support-v4:27.1.1'
```
Comment (pfent) [inline app/build.gradle]: support-v4 shouldn't be needed

**Comment by velev8** [comment]
Fixed the grey bar / landscape orientation by locking portrait orientation mode during the QR Scanner

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/TicketScanValidationActivity.java (hunk 1 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/TicketScanValidationActivity.java]: @saileru is right. Sorry, my bad :man_shrugging:

**Pull request merged**